### PR TITLE
Category use case tests added and docstring added

### PR DESCRIPTION
**Pull request by EmilyDo**
Created Use Case Interactor tests (haven't actually tested everything yet) and docstring for all the Category files (that aren't UI). 

Design Patterns: Used private class pattern so other classes can only access attributes through getters and setters

- [x]  Unit tests (For task feature): Before PR: 10% After PR: 50%
- [ ]  Exception Handling: Exceptions mostly handled.
- [x]  JavaDocs Documentation
- [x]  SOLID principles
- [x]  Clean Architecture

### CODE REVIEW

**Review by m-banthwan — APPROVED** [review]
Good Java docs!

**Review by MadhavKanna — APPROVED** [review]
Approved, everything is quality!

**Review by JoAndNight — DISMISSED** [review]

**Review by JoAndNight — APPROVED** [review]

File: src/main/java/ui/category/ToDoViewModel.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 import entities.CommonUser;
 import use_cases.categories.create_category.CreateCategoryOutputData;
 
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.ActionListener;
 import java.util.ArrayList;
 
+*/
+/**
```
Comment (MadhavKanna) [inline src/main/java/ui/category/ToDoViewModel.java:14]: Do implement the full ToDoViewModel in the future

**Review by MadhavKanna — APPROVED** [review]
Approved

**Pull request merged**